src/main/scala/analysis/Local.scala

```diff
@@ -0,0 +1,351 @@
+package analysis
+
+import analysis.BitVectorEval.{bv2SignedInt, isNegative}
+import ir.{Assign, BVADD, BinaryExpr, BitVecLiteral, BitVecType, CFGPosition, DirectCall, Endian, Expr, Extract, IntraProcIRCursor, MemoryAssign, MemoryLoad, Procedure, Register, Variable, ZeroExtend, computeDomain, toShortString}
+import specification.{ExternalFunction, SpecGlobal, SymbolTableEntry}
+
+import scala.util.control.Breaks.{break, breakable}
+import java.math.BigInteger
+import scala.collection.mutable
+
+/**
+ * The local phase of Data Structure Analysis
+ * @param proc procedure to be analysed
+ * @param symResults result of symbolic access analysis
+ * @param constProp
+ * @param globals
+ * @param globalOffsets
+ * @param externalFunctions
+ * @param reachingDefs
+ * @param writesTo mapping from procedures to registers they change
+ * @param params mapping from procedures to their parameters
+ */
+class Local(
+             proc: Procedure,
+             symResults: Map[CFGPosition, Map[SymbolicAccess, TwoElement]],
+             constProp: Map[CFGPosition, Map[Variable, FlatElement[BitVecLiteral]]],
+             globals: Set[SymbolTableEntry], globalOffsets: Map[BigInt, BigInt],
+             externalFunctions: Set[ExternalFunction],
+             reachingDefs: Map[CFGPosition, Map[Variable, Set[CFGPosition]]],
+             writesTo: Map[Procedure, Set[Register]],
+             params:  Map[Procedure, Set[Variable]]
+           ) extends Analysis[Any]{
+
+  private val mallocRegister = Register("R0", 64)
+  private val stackPointer = Register("R31", 64)
+
+  // set of cfg positions already processed by the analysis local phase
+  private val visited: mutable.Set[CFGPosition] = mutable.Set()
+
+
+  // variables to symbolic access map for each cfg position
+  val varToSym: Map[CFGPosition, Map[Variable, Set[SymbolicAccess]]] = symResults.foldLeft(Map[CFGPosition, Map[Variable, Set[SymbolicAccess]]]()) {
+    (outerMap, syms) =>
+      val position = syms._1
+      val innerMap = syms._2.foldLeft(Map[Variable, Set[SymbolicAccess]]()) {
+        (m, access) =>
+          if m.contains(access._1.accessor) then
+            // every variable pointing to a stack region ONLY has one symbolic access associated with it.
+            m(access._1.accessor).foreach(
+              sym => assert(!sym.symbolicBase.isInstanceOf[StackLocation])
+            )
+            assert(!access._1.symbolicBase.isInstanceOf[StackLocation])
```

Comment:
This assertion fails for the input examples\indirect_call_outparam\indirect_call_outparam.adt

Reply:
These assertions just don't seem to hold up much in practice, what's the assumption behind them? There seems to be an issue with it